src/main/java/org/jenkinsci/plugins/ghprb/extensions/build/GhprbCancelBuildsOnUpdate.java

```diff
@@ -39,15 +39,15 @@ public Boolean getOverrideGlobal() {
         return overrideGlobal == null ? false : overrideGlobal;
     }
 
-    private void cancelCurrentBuilds(AbstractProject<?, ?> project,
+    private void cancelCurrentBuilds(Job<?, ?> project,
                                      Integer prId) {
         if (getOverrideGlobal()) {
             return;
         }
         
         logger.log(Level.FINER, "New build scheduled for " + project.getName() + " on PR # " + prId + ", checking for queued items to cancel.");
         Queue queue = Jenkins.getInstance().getQueue();
-        for (Queue.Item item : queue.getItems(project)) {
+        for (Queue.Item item : queue.getItems()) {
```

Comment:
@ButkiewiczP There might be more than one, right.  Actually, looking again this code is probably not correct at all. IIRC, The build scheduling happens in multiple threads, so this means it could theoretically cancel builds just submitted.

Maybe the right thing to do is to call the getQueueItem() in a loop and cancel until there are no items left.  This still has a bit of a race condition (let's say you do 'test this please' twice in rapid succession) but maybe that's less of an actual issue.